### PR TITLE
feat: enable vcpkg caching in buildpacks

### PR DESCRIPTION
**Pull request by coryan**
This change is [Reviewable](https://reviewable.io/reviews/googlecloudplatform/functions-framework-cpp/169)

### CODE REVIEW

**Comment by coryan** [comment]
/cc: @squee1945 FYI, now with caching for the dependencies.

**Review by squee1945 — CHANGES_REQUESTED** [review]
I think these hard-coded /layers references need to change?

**Comment by coryan** [comment]
> I think these hard-coded /layers references need to change?

You are right. The directories are part of the base image though, I will change that in a separate PR, moving them to `/usr/local/` or something similar.

**Review by devjgm — APPROVED** [review]

**Comment by coryan** [comment]
@squee1945 any thoughts?

File: pack/buildpack/bin/build [hunk 1]
```diff
@@ -33,20 +33,37 @@ launch = false
...
-cp -r /workspace /var/tmp/source/application
-test -r /workspace/vcpkg.json && cp /workspace/vcpkg.json /var/tmp/source
-/layers/cpp/gcf/generate-wrapper.sh "${TARGET_FUNCTION}" "${FUNCTION_SIGNATURE_TYPE:-http}" >/var/tmp/source/main.cc
+cat > "${layers}/source.toml" <<EOL
+build = true
+cache = false
+launch = false
+EOL
+
+cp -r /layers/cpp/gcf/cmake "${layers}/source"
```
Comment (squee1945) [inline pack/buildpack/bin/build:42]: More `/layers` to convert?

**Review by squee1945 — APPROVED** [review]
On the migration to GCP Buildpacks, I suspect we'll want to separate this into multiple buildpacks that can be composed into difference target scenarios (in particular, OSS uber-builder vs. GCF runtime-specific builder).

File: pack/buildpack/bin/build (hunk 1 above)
Comment (coryan) [reply pack/buildpack/bin/build:42]: Yup, thanks.

**Pull request merged**